bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdLanguageServerProvider.java

```diff
@@ -10,44 +10,26 @@
  * Contributors:
  * Gesa Hentschke (Bachmann electronic GmbH) - initial implementation
  * Alexander Fedorov (ArSysOp) - rework to OSGi components
+ * Alexander Fedorov (ArSysOp) - rework access to preferences
  *******************************************************************************/
 
 package org.eclipse.cdt.lsp.internal.clangd;
 
 import java.net.URI;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.List;
 import java.util.Objects;
 
 import org.eclipse.cdt.lsp.clangd.BaseClangdLanguageServerProvider;
+import org.eclipse.cdt.lsp.clangd.ClangdConfiguration;
 import org.eclipse.cdt.lsp.clangd.ClangdFallbackFlags;
-import org.eclipse.cdt.lsp.internal.clangd.editor.LspEditorUiPlugin;
-import org.eclipse.cdt.lsp.internal.clangd.editor.preferences.LspEditorPreferences;
-import org.eclipse.cdt.utils.CommandLineUtil;
 import org.eclipse.core.resources.IProject;
-import org.eclipse.core.resources.ProjectScope;
-import org.eclipse.core.runtime.Platform;
 import org.eclipse.core.runtime.ServiceCaller;
-import org.eclipse.core.runtime.preferences.IScopeContext;
-import org.eclipse.core.runtime.preferences.PreferenceMetadata;
-import org.eclipse.jface.preference.IPreferenceStore;
 
-public class ClangdLanguageServerProvider extends BaseClangdLanguageServerProvider {
-	//FIXME: AF: rework to core preferences
-	private static final IPreferenceStore preferenceStore = LspEditorUiPlugin.getDefault().getLsPreferences();
-	private static final PreferenceMetadata<Boolean> option = LspEditorPreferences.getPreferenceMetadata();
+public final class ClangdLanguageServerProvider extends BaseClangdLanguageServerProvider {
 
-	@Override
-	protected List<String> createCommands() {
-		List<String> commands = super.createCommands();
-		setPreferenceStoreDefaults(commands); // use the server provider settings as default
-		List<String> commandsFromStore = getCommandsFromStore();
-		if (commandsFromStore.isEmpty()) {
-			return commands;
-		}
-		return commandsFromStore;
-	}
+	private final ServiceCaller<ClangdConfiguration> configuration = new ServiceCaller<>(getClass(),
+			ClangdConfiguration.class);
 
```

Comment:
What happens if a vendor adds his own `ClangdConfiguration`? How is ensured, that `ClangdConfigurationAccess` will be used here?

Reply:
This PR is focused at reworking access to preferences. We have a separate ticket #81 to work on configuration customization. I suggest to discuss the customization points and approaches there.
For example, we can add "server id" metadata to both "provider" and "configuration" and then invoke `ServiceCaller` with specific `filter` argument. Or think about something else.